### PR TITLE
gcylc right-click menu specificity

### PR DESCRIPTION
**Pull request by arjclark**
Addresses #1321 

Trigger (now and edit), poll and kill right-click menu options are now enabled/disabled based on the task state.

@hjoliver - please review 1
@kaday - please review 2

I've played around with a running suite to check things are working but you'll probably want to do some manual testing yourselves. Please also double-check I've covered the right states for these actions.

### CODE REVIEW

**Comment by kaday** [comment]
tests/pep8/00-bin-lib.t - Failed tests:  1-2

File: lib/cylc/gui/app_gcylc.py [hunk 1]
```diff
@@ -509,6 +509,29 @@ class ControlApp(object):
...
+
+    STATES_KILLABLE = [
+        'submitted',
+        'running'
+    ]
+
+    STATES_POLLABLE = [
+        'submitted',
+        'running'
+    ]
```
Comment (hjoliver) [inline lib/cylc/gui/app_gcylc.py]: I wonder if we should make 'succeeded' and 'failed' pollable too?  The poll would consult the task status file, and would therefore distinguish between tasks that really finished, and those whose states were manually reset to succeed or failed... which might occasionally be useful.

**Comment by hjoliver** [comment]
"Trigger (edit run)" should be grayed out for active tasks, like "Trigger (run now)".

**Comment by hjoliver** [comment]
We should probably also make "View -> job script/out/err" inactive for tasks that have not yet started running.  Currently you can select that, but will get a "task has not started yet" warning dialog.

**Comment by hjoliver** [comment]
As per Kerry's comment:
```
cylc test-b tests/pep8/
tests/pep8/00-bin-lib.t .. 1/3 --- 00-bin-lib.stdout	2016-01-05 09:39:31.978098580 +0000
+++ -	2016-01-05 09:39:32.646910846 +0000
@@ -1,6 +0,0 @@
-lib/cylc/gui/app_gcylc.py:535:5: E303 too many blank lines (2)
-lib/cylc/gui/app_gcylc.py:1209:80: E501 line too long (87 > 79 characters)
-lib/cylc/gui/app_gcylc.py:1216:80: E501 line too long (82 > 79 characters)
-lib/cylc/gui/view_dot.py:111:25: E126 continuation line over-indented for hanging indent
-lib/cylc/gui/view_graph.py:193:1: W293 blank line contains whitespace
-lib/cylc/gui/view_tree.py:183:25: E126 continuation line over-indented for hanging indent
```

**Comment by arjclark** [comment]
@hjoliver @kaday - back from holiday now, sorry for the delay. 

Feedback addressed - sorry about the pep8, made a last minute change and forgot to re-run the test.

The trigger edit run should now enable disable accordingly, I've enabled polling for succeeded/failed tasks as requested and I've set sensitivity for the task view menus. Again I've given it a run through the various views but please test in vain to make sure I've not broken anything by accident.

**Comment by arjclark** [comment]
@hjoliver - now you're back are you able to sign off on this?

**Comment by hjoliver** [comment]
(After 1.5 days back at work, I had to give a full day workshop today - but I'm back on deck tomorrow!)

**Comment by arjclark** [comment]
Cool, ta.

**Comment by hjoliver** [comment]
@arjclark - all good, except for this:

>  I've enabled polling for succeeded/failed tasks as requested

This also needs to be enabled in the suite daemon:
```
cylc log states | grep poll
2016-02-02T08:32:42+13 INFO - [client-command] poll_tasks oliverh@niwa-34403.niwa.local:cylc-gui 30bcd6b9-4636-4437-b4f9-cd7962d74660
2016-02-02T08:32:43+13 WARNING - cfoo.20120808T0000+13: skip poll, state not ["submitted", "running"]
2016-02-02T08:32:43+13 INFO - Command succeeded: poll_tasks(cfoo,20120808T0000+13,False)
```

**Comment by arjclark** [comment]
> This also needs to be enabled in the suite daemon

@hjoliver - hmm, I hadn't appreciated that wasn't already enabled. I've now put the sensitivity back to that of only being active in the submitted or running states.

If we want the capability to poll the other states can we do that in a separate ticket? My reasoning being that some batch scheduler setups can react badly to repeated calls to their status command(s) so enabling polling of the final states being pollable in the daemon could be potentially hazardous to the scheduler in a large suite such as our operational models if someone were to hit the "poll all" button.

**Comment by hjoliver** [comment]
@arjclark - good point, agreed.  In which case: Review 1, all good.

**Comment by kaday** [comment]
Looks good to me.
Test battery passes as expected in my environment.
gcylc behaves as expected.

**Pull request merged**